softgroup/basis/common-router/common-router-api/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java

```diff
@@ -1,20 +1,36 @@
 package com.softgroup.common.router.api;
 
 
-import com.softgroup.common.protocol.Request;
-import com.softgroup.common.protocol.RequestData;
-import com.softgroup.common.protocol.Response;
-import com.softgroup.common.protocol.ResponseData;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.softgroup.common.datamapper.DataMapper;
+import com.softgroup.common.exceptions.MapperException;
+import com.softgroup.common.protocol.*;
+import org.springframework.beans.factory.annotation.Autowired;
 
-public class AbstractRequestHandler<T extends RequestData, R extends ResponseData> implements RequestHandler {
-	@Override
-	public String getName() {
-		return null;
-	}
+
+public abstract class AbstractRequestHandler<T extends RequestData, R extends ResponseData> implements RequestHandler {
+
+	@Autowired
+	private DataMapper mapper;
 
 	@Override
 	public Response<R> handle(Request<?> msg) {
-		return null;
+		try {
+			T data = mapper.convert(msg.getData(), new TypeReference<T>() {});
+			Request<T> castedRequest = new Request.Builder<T>()
```

Comment:
@melnyknikolay Good idea would be to make request/responce factory for such general cases.